src/Functions/FunctionsCodingUUID.cpp

```diff
@@ -13,43 +13,160 @@
+    };
+
+    explicit UUIDSerializer(const Variant variant)
+        : first_half_binary_representation(variant == Variant::Microsoft ? Representation::LittleEndian : Representation::BigEndian)
+    {
+        if (variant != Variant::Default && variant != Variant::Microsoft)
+            throw DB::Exception(DB::ErrorCodes::LOGICAL_ERROR, "{} is not handled yet", magic_enum::enum_name(variant));
+    }
+
+    void deserialize(const UInt8 * src16, UInt8 * dst36) const
+    {
+        formatHex({src16, 4}, &dst36[0], first_half_binary_representation);
+        dst36[8] = '-';
+        formatHex({src16 + 4, 2}, &dst36[9], first_half_binary_representation);
+        dst36[13] = '-';
+        formatHex({src16 + 6, 2}, &dst36[14], first_half_binary_representation);
+        dst36[18] = '-';
+        formatHex({src16 + 8, 2}, &dst36[19], Representation::BigEndian);
+        dst36[23] = '-';
+        formatHex({src16 + 10, 6}, &dst36[24], Representation::BigEndian);
+    }
+
+    void serialize(const UInt8 * src36, UInt8 * dst16) const
+    {
+        /// If string is not like UUID - implementation specific behaviour.
+        parseHex(&src36[0], {dst16 + 0, 4}, first_half_binary_representation);
+        parseHex(&src36[9], {dst16 + 4, 2}, first_half_binary_representation);
+        parseHex(&src36[14], {dst16 + 6, 2}, first_half_binary_representation);
+        parseHex(&src36[19], {dst16 + 8, 2}, Representation::BigEndian);
+        parseHex(&src36[24], {dst16 + 10, 6}, Representation::BigEndian);
+    }
+
+private:
+    Representation first_half_binary_representation;
+};
+
+void checkArgumentCount(const DB::DataTypes & arguments, const std::string_view function_name)
 {
+    if (const auto argument_count = std::ssize(arguments); argument_count < 1 || argument_count > 2)
+        throw DB::Exception(
+            DB::ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+            "Number of arguments for function {} doesn't match: passed {}, should be 1 or 2",
+            function_name,
+            argument_count);
+}
 
-namespace ErrorCodes
+void checkFormatArgument(const DB::DataTypes & arguments, const std::string_view function_name)
 {
-    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
-    extern const int ILLEGAL_COLUMN;
+    if (const auto argument_count = std::ssize(arguments);
+        argument_count > 1 && !DB::WhichDataType(arguments[1]).isInt8() && !DB::WhichDataType(arguments[1]).isUInt8())
+        throw DB::Exception(
+            DB::ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+            "Illegal type {} of second argument of function {}, expected Int8 or UInt8 type",
+            arguments[1]->getName(),
+            function_name);
+}
+
+UUIDSerializer::Variant parseVariant(const DB::ColumnsWithTypeAndName & arguments)
+{
+    if (arguments.size() < 2)
+        return UUIDSerializer::Variant::Default;
+
+    const auto representation = static_cast<magic_enum::underlying_type_t<UUIDSerializer::Variant>>(arguments[1].column->getInt(0));
```

Comment:
> `arguments[1].column->getInt(0)`

AFAIK, the column may be empty (especially when called from `executeDryRunImpl`), so it may crash:
https://s3.amazonaws.com/clickhouse-test-reports/61799/ace983f3d5d20c4cfaf6220cda7e3251bfb93278/ast_fuzzer__msan_.html

msan report: https://pastila.nl/?000d758a/f6f27b43f714b735c5b0bb5246f3531c#DdJW2oeYhxwNOiJON6ioOQ==